src/transformers/models/t5/modeling_t5.py

```diff
@@ -647,8 +649,9 @@ def forward(
         )
         hidden_states, present_key_value_state = self_attention_outputs[:2]
         attention_outputs = self_attention_outputs[2:]  # Keep self-attention outputs and relative position weights
-
+        
         # clamp inf values to enable fp16 training
+        #TODO
         if hidden_states.dtype == torch.float16 and torch.isinf(hidden_states).any():
```

Comment:
@stas00 like the prior question, is `hidden_states.dtype` supposed to be 16-bit when amp is enabled?  I was under the impression that all variables would remain represented in float-32 except when in a matrix operation (could be wrong!)

Reply:
(my own print statements also are saying that hidden states are float32, even when autocast is enabled)

Reply:
I replied above. https://github.com/huggingface/transformers/pull/13207#discussion_r694454685

it is quite confusing, I get confused quite often about this.